gprofiler/profilers/java.py

```diff
@@ -676,11 +685,16 @@ def _check_async_profiler_loaded(self, process: Process) -> bool:
         return False
 
     def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
```

Comment:
Return value is no longer `Optional`

Reply:
Yup. Interface is still Optional, but will change it here

Reply:
Why? what other runtimes have elective disablement? Should change the interface and do the same for everything IMO

Reply:
Uh sadly not precisely:
```
        if output is None:
            logger.warning(f"Profiled process {ap_proc.process.pid} exited before reading the output")
            return None
```

Some of these I have fixed in a later PR. Will change it a future PR.

Reply:
> Why? what other runtimes have elective disablement? Should change the interface and do the same for everything IMO

Not selective disablement, but other failure reasons, as I have found here.

Reply:
Well - we do want all of those to also not just have native stacks, but an "error" stack, but if you want to delay to a later PR, it is OK